core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/WorkflowResource.scala

```diff
@@ -104,6 +104,32 @@ class WorkflowResource {
     }
   }
 
+  /**
+    * This method creates and insert a new workflow to database
+    *
+    * @param session  HttpSession
+    * @param workflow , a workflow
+    * @return Workflow, which contains the generated wid if not provided
+    */
+  @POST
+  @Path("/create")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def createWorkflow(@Session session: HttpSession, workflow: Workflow): Response = {
+    UserResource.getUser(session) match {
+      case Some(user) =>
+        if (workflowOfUserExists(workflow.getWid, user.getUid)) {
+          Response.status(Response.Status.BAD_REQUEST).build()
```

Comment:
when creating a new workflow, the frontend should send a workflow without id.  The id is being generated in db only. So the logic here should be checking if the wid is null then insert it. If wid is provided in the argument, it should reject the request.

Reply:
Fixed, will be updated in the next commit